### PR TITLE
chore: backport latest changes from original repo

### PR DESCRIPTION
**Pull request by Afoucaul**
## Description

Backport latest changes from https://github.com/CardioLogs/validate-github-actions

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -41,7 +46,12 @@ runs:
     - name: Run actionlint and format output as JSON
       id: actionlint
       shell: bash
-      run: actionlint -format '{{json .}}' | tee actionlint_output.json
+      run: |
+        # Reference: https://github.com/actions/download-artifact#outputs
+        actionlint \
+          -ignore 'property "download-path" is not defined in object type {}' \
+          -format '{{json .}}' \
+        | tee actionlint_output.json
```
Comment (DidelotK) [inline action.yml:73]: You can use `run: >` and get rid of all the backslashes, cf https://yaml-multiline.info/

Note that the comment line has to go outside the `run` block then.

File: action.yml [hunk 2]
```diff
@@ -19,13 +22,15 @@ runs:
...
+        labels = "${{ inputs.self-hosted-runners }}".split(",")
+        config = f"""\
         self-hosted-runner:
-          labels:
-            - aws-gpu-runner
-        EOF
+          labels: {labels}
+        """
+        with open(".github/actionlint.yaml", "w") as fd:
+            fd.write(config)
```
Comment (DidelotK) [inline action.yml]: What if the file already exists?
What if the file already has the `self-hosted-runner.labels` key?
What if `self-hosted-runner.labels` is already filled?

File: action.yml
Comment (DidelotK) [inline action.yml:1]: (I can't comment [this line](https://github.com/philips-software/github-actions-linter/pull/1/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R41) as it's not modified)

```
41:     excludes="-e $(echo ${{ inputs.shellcheck-disable-codes }} | sed -e 's/^,//' -e 's/,$//' -e 's/,/ -e /g')"
```

Can we add a comment explaining the inputs and outputs of this (I guess an example is enough)? The `-e` in both _sed_ arguments and commands are quite confusing.

File: action.yml [hunk 3]
```diff
@@ -19,13 +22,31 @@ runs:
...
+            with config_path.open() as fd:
+                return yaml.safe_load(fd)
+
+        def save_config(config):
+            with config_path.open("w") as fd:
+                yaml.safe_dump(config, fd)
+
+        config = get_config()
+        labels = "${{ inputs.self-hosted-runners }}".split(",")
+        config.setdefault("self-hosted-runner", {}).setdefault("labels", []).extend(labels)
```
Comment (DidelotK) [inline action.yml:44]: Yay, so Pythonic. I love it.

Now, how does ActionLint react if `self-hosted-runner.labels` contains multiple times the same entry?

File: action.yml (hunk 1 above)
Comment (Afoucaul) [reply action.yml:73]: This is a backport so I'm keeping it as is for now

File: action.yml (hunk 2 above)
Comment (Afoucaul) [reply action.yml]: I've rewritten the Python script to account for that

File: action.yml
Comment (Afoucaul) [reply action.yml:1]: Added a comment

File: action.yml (hunk 3 above)
Comment (Afoucaul) [reply action.yml:44]: deduplicated labels

**Review by DidelotK — APPROVED** [review]
Nice, LGTM :)

**Pull request merged**